opensrp-chw-core/src/main/java/org/smartregister/chw/core/dao/StockUsageReportDao.java

```diff
@@ -0,0 +1,120 @@
+                "  AND vdd.details LIKE '%condom' \n" +
+                "  group by vdd.details, strftime('%Y',(datetime(v.visit_date/1000, 'unixepoch', 'localtime'))), strftime('%m',(datetime(v.visit_date/1000, 'unixepoch', 'localtime')))\n" +
+                " UNION ALL\n" +
+                " SELECT substr(fp.fp_start_date, 7, 4) as \"Year\",\n" +
+                "\tsubstr(fp.fp_start_date, 4, 2) as \"Month\",\n" +
+                "\tfp.fp_method_accepted as StockName,\n" +
+                "\tcount(1) as Usage\n" +
+                " FROM ec_family_planning fp\n" +
+                "  WHERE ( fp.fp_method_accepted like 'Standard day method' or fp.fp_method_accepted like 'Emergency contraceptive')\n" +
+                "  AND fp.fp_start_date is not NULL\n" +
+                "  group by fp.fp_method_accepted, substr(fp.fp_start_date, 7, 4), substr(fp.fp_start_date, 4, 2)\n" +
+                "UNION ALL\n" +
+                "SELECT \n" +
+                " substr(mc.malaria_test_date, 7, 4) as Year,\n" +
+                " substr(mc.malaria_test_date, 4, 2) as Month,\n" +
+                " mc.malaria_treat as StockName,\n" +
+                " count(1) as Usage\n" +
+                "\tfrom ec_malaria_confirmation mc\n" +
+                "where mc.malaria_treat in('ALU 6','ALU 12', 'ALU 18', 'ALU 24')\n" +
+                "  and mc.malaria_test_date is not NULL\n" +
+                "  group by mc.malaria_treat, substr(mc.malaria_test_date, 7, 4), substr(mc.malaria_test_date, 4, 2)\n" +
+                " UNION ALL\n" +
+                " SELECT substr(mc.malaria_test_date, 7, 4) as \"Year\",\n" +
+                "        substr(mc.malaria_test_date, 4, 2) as \"Month\",\n" +
+                "\t\t'RDTs' as StockName,\n" +
+                "\t\tcount(1) as Usage\n" +
+                "\tfrom ec_malaria_confirmation mc\n" +
+                "where mc.malaria_results in('Negative','Positive')\n" +
+                "and mc.malaria_test_date is not NULL\n" +
+                "  group by mc.malaria_treat, substr(mc.malaria_test_date, 7, 4), substr(mc.malaria_test_date, 4, 2)\n" +
+                "order by Year DESC, Month DESC";
+
+        DataMap<StockUsage> dataMap = cursor -> {
+            StockUsage stockUsage = new StockUsage();
+            stockUsage.setYear(getCursorValue(cursor, "Year"));
+            stockUsage.setMonth(getCursorValue(cursor, "Month"));
+            stockUsage.setStockName(getCursorValue(cursor, "StockName"));
+            stockUsage.setStockUsage(getCursorValue(cursor, "Usage"));
+            return stockUsage;
+        };
+
+        List<StockUsage> res = readData(sql, dataMap);
+        if (res == null)
+            return new ArrayList<>();
+
+        return res;
+    }
+
+    /**
+     * Reads the min last update date and compares to today's date
+     * if last update date is not in same month as today return false
+     *
+     * @return
+     */
+    public static boolean lastInteractedWithinDay() {
```

Comment:
Does this check whether the `updated_at` date is withing the same month or same day? 🤔